xdsl/__init__.py

```diff
@@ -19,7 +19,7 @@ def load_mlir_module(mlir_module: _xdsl_init_types.ModuleType) -> None:
     _REQUIRED_MLIR_MODULES by the caller prior to calling this function.
     """
     global _mlir_module
-    if _mlir_module != mlir_module:
+    if _mlir_module and _mlir_module != mlir_module:
```

Comment:
Maybe you can add a short comment along the lines of the PR description? Otherwise GTG

Reply:
The intro comment of the function describes some of this behavior already. I can add the case destinction in the PR description.